summarization-api-playground/src/main.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './style.css'
+
+const inputTextArea = document.querySelector('#input') as HTMLTextAreaElement;
+const characterCountSpan = document.querySelector('#character-count') as HTMLSpanElement;
+const summarizationUnsupportedDialog = document.querySelector('#summarization-unsupported') as HTMLDialogElement;
+const summarizationUnavailableDialog = document.querySelector('#summarization-unavailable') as HTMLDialogElement;
+const output = document.querySelector('#output') as HTMLDivElement;
+
+/*
+ * Creates a summarization session. If the model has already been downloaded, this function will
+ * create the session and return it. If the model needs to be downloaded, the this function will
+ * wait for the download to finish before resolving the promise.
+ * 
+ * If a downloadProgressCallback is provided, the function will add the callback to the session
+ * creation.
+ * 
+ * The function expects the model availability to be either `readily` or `after-download`, so the
+ * availability must be checked before calling it. If availabilityis `no`, the function will throw
+ *  an error.
+ */
+const createSummarizationSession = async (downloadProgressCallback?: AIModelDownloadCallback): Promise<AISummarizerSession> =>  {
+  const canSummarize = await window.ai.summarizer!.capabilities();
+  if (canSummarize.available == 'no') {
```

Comment:
```suggestion
  if (canSummarize.available === 'no') {
```

Reply:
TypeScript does warn for comparison of different types using `==`, but changing for consistency.

Reply:
Updated